Formula/tenyr.rb

```diff
@@ -14,43 +14,53 @@ class Tenyr < Formula
     sha256 "c20b4f63feb6150edc2e46222eed8329d0e8e5bb8bdfa3f84a87059ed03d95a4" => :mavericks
   end
 
+  # pkg-config is used after v0.9.3 for sdl2, instead of sdl2-config
+  # prepare for post-v0.9.3 versions (including HEAD) by depending on it now
+  depends_on "pkg-config" => :build
   depends_on "bison" => :build # tenyr requires bison >= 2.5
-  depends_on "sdl2"
-  depends_on "sdl2_image"
-
-  patch :DATA
+  # sdl2_image implies sdl2. If we specify sdl2 separately, we create
+  # nonsensical possibilities like `--with-sdl2_image --without-sdl2`
+  # tenyr requires sdl2_image --with-png
+  depends_on "sdl2_image" => :recommended
 
   def install
     bison = Formula["bison"].bin/"bison"
 
-    inreplace "Makefile", "bison", bison
+    args = []
 
-    system "make"
-    bin.install "tsim", "tas", "tld"
-  end
-end
+    # specify our own bison, since we need bison >= 2.5
+    args << "BISON=" + bison
 
-__END__
-diff --git a/src/debugger_parser.y b/src/debugger_parser.y
-index c000640..289e4da 100644
---- a/src/debugger_parser.y
-+++ b/src/debugger_parser.y
-@@ -13,6 +13,7 @@
- int tdbg_error(YYLTYPE *locp, struct debugger_data *dd, const char *s);
+    # JIT build is not available until we can pull in AsmJit somehow
+    # HEAD version can build with JIT enabled, using git submodule
+    # Right now there is no way for `build.with?("jit")` to be true
+    if build.without?("jit")
+      args << "JIT=0"
+    end
 
- #define YYLEX_PARAM (dd->scanner)
-+void *yyscanner;
+    # Use our own build directory (tenyr's default build directory encodes
+    # builder platform information in the path)
+    builddir = "build/homebrew"
+    args << "BUILDDIR=" + builddir
 
- %}
+    if build.without?("sdl2_image")
+      args << "SDL=0"
+    end
 
-diff --git a/src/parser.y b/src/parser.y
-index 6eb4dd1..e3f23df 100644
---- a/src/parser.y
-+++ b/src/parser.y
-@@ -40,6 +40,7 @@ static int check_immediate_size(struct parse_data *pd, YYLTYPE *locp, uint32_t
-         imm);
+    system "make", *args
 
- #define YYLEX_PARAM (pd->scanner)
-+void *yyscanner;
+    pkgshare.install "rsrc", "plugins"
+    cd builddir do
+      bin.install "tsim", "tas", "tld"
+      lib.install Dir["*.dylib"]
+    end
+  end
 
- struct symbol *symbol_find(struct symbol_list *list, const char *name);
+  test do
+    # sanity test assembler, linker and simulator
+    `echo 'B <- 9'     | tas --output=a.to -`
```

Comment:
These should probably be wrapped with `system` instead, and file creation should be isolated:

```ruby
(testpath/"part1").write "B <- 9"
(testpath/"part2").write "C <- B * 3"

system "tas", "--output=a.to", "part1"
system "tas", "--output=b.to", "part2"
 # etc...

assert_match "C 0000001b", `tsim -vvvv test.texe`
```